### PR TITLE
Pitch teacher

### PR DESCRIPTION
**Pull request by duerig**
Revert to old paradigm of an acoustic teacher model for pitch/energy style. Add a new kind of duration loss function.

### CODE REVIEW

**Comment by Kreevoz** [comment]
I'm observing the changes you make to your codebase with quite some curiosity. Would you say that you've hit the synthesis quality you were after, or are you still experimenting? Just wondering when would be a good time to give this a try and train a model or two on it.

**Comment by duerig** [comment]
The quality is very good in my experiments. The current roadblock is exporting from torch. There is an odd problem where the ONNX version of the model introduces a metallic-sounding artifact. Something similar happened to Kokoro to a lesser extent. Other export options (tflite, tvm) have their own road blocks that I discovered.

Once I get this resolved I'll be able to focus on cleaning up code and making it easier to use.

If you have ideas or want to discuss things more, I am active as Tallow on the Kokoro and StyleTTS 2 discords and if you PM me I can invite you to the Stylish discord which is not yet public.

**Comment by Kreevoz** [comment]
That Stylish discord sounds useful. I'll hit you up for that.
